### PR TITLE
Add blueprint resource and data source (#99) (#104)

### PR DESCRIPTION
**Pull request by dmettem**
Signed-off-by: Deepak Mettem <dmettem@vmware.com>

### CODE REVIEW

File: vra/data_source_blueprint.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+	if projectIdOk {
+		projects = append(projects, projectId.(string))
+		resp, err = apiClient.Blueprint.ListBlueprintsUsingGET1(
+			blueprint.NewListBlueprintsUsingGET1Params().WithName(withString(name.(string))).WithProjects(projects))
+	} else {
+		resp, err = apiClient.Blueprint.ListBlueprintsUsingGET1(
+			blueprint.NewListBlueprintsUsingGET1Params().WithName(withString(name.(string))))
+	}
+	log.Printf("After get all call")
```
Comment (markpeek) [inline vra/data_source_blueprint.go]: Remove debug printf

File: vra/data_source_blueprint.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+		d.Set("total_released_versions", blueprint.TotalReleasedVersions)
+		d.Set("total_versions", blueprint.TotalVersions)
+		d.Set("updated_at", blueprint.UpdatedAt)
+		d.Set("updated_by", blueprint.UpdatedBy)
+		d.Set("valid", blueprint.Valid)
+	}
+
+	for _, bp := range resp.Payload.Content {
+		if (idOk && bp.ID == id) || (nameOk && bp.Name == name.(string)) {
+			log.Printf("Before making the GET on id call")
```
Comment (markpeek) [inline vra/data_source_blueprint.go]: Remove debug printf

File: vra/resource_blueprint.go [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+	d.Set("total_versions", blueprint.TotalVersions)
+	d.Set("updated_at", blueprint.UpdatedAt)
+	d.Set("updated_by", blueprint.UpdatedBy)
+	d.Set("valid", blueprint.Valid)
+
+	if err := d.Set("validation_messages", flattenValidationMessages(blueprint.ValidationMessages)); err != nil {
+		return fmt.Errorf("error setting validation_messages in blueprint - error: %#v", err)
+	}
+
+	log.Printf("finished reading the vra_blueprint resource with name %s", d.Get("name"))
```
Comment (markpeek) [inline vra/resource_blueprint.go]: Please review the printf's in this file. If keeping them, perhaps capitalize "finished"?

File: vra/data_source_blueprint.go (hunk 1 above)
Comment (dmettem) [reply vra/data_source_blueprint.go]: Done.

File: vra/data_source_blueprint.go (hunk 2 above)
Comment (dmettem) [reply vra/data_source_blueprint.go]: Done.

File: vra/resource_blueprint.go (hunk 3 above)
Comment (dmettem) [reply vra/resource_blueprint.go]: Done.

**Pull request merged**